catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/activity/rtl/RtlLayoutsTest.java

```diff
@@ -0,0 +1,247 @@
+
+@RunWith(AndroidJUnit4.class)
+public class RtlLayoutsTest {
+	private Locale arabicLocale = new Locale("ar");
+	private Configuration config = getResources().getConfiguration();
+	private final String title = getResourcesString(R.string.main_menu_scratch_converter);
+	private final String beta = getResourcesString(R.string.beta).toUpperCase(Locale.getDefault());
+	private final String mainMenuScratchConverterBeta = title + " " + beta;
+	@Rule
+	public BaseActivityInstrumentationRule<MainMenuActivity> baseActivityTestRule = new BaseActivityInstrumentationRule<>(MainMenuActivity.class,
+			true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		//delete all SharedPreferences to avoid "ShowDetails/HideDetails" problem
+		//you will face the described Problem when you run this test more than one time respectively
+		File root = InstrumentationRegistry.getTargetContext().getFilesDir().getParentFile();
+		String[] sharedPreferencesFileNames = new File(root, "shared_prefs").list();
+		for (String fileName : sharedPreferencesFileNames) {
+			InstrumentationRegistry.getTargetContext().getSharedPreferences(fileName.replace(".xml", ""),
+					Context.MODE_PRIVATE).edit().clear().commit();
+		}
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN_MR1)
+	@Test
+	public void assertLayoutsAndTextDirectionAreRTL() {
+		assertEquals(View.LAYOUT_DIRECTION_RTL, config.getLayoutDirection());
+		assertEquals(arabicLocale.getDisplayLanguage(), Locale.getDefault().getDisplayLanguage());
+
+		onView(withId(R.id.main_menu_button_continue))
+				.check(matches(isDisplayed()));
+		onView(withId(R.id.main_menu_button_new))
+				.check(matches(isDisplayed()));
+		onView(withId(R.id.main_menu_button_programs))
+				.check(matches(isDisplayed()));
+		onView(withId(R.id.main_menu_button_help))
+				.check(matches(isDisplayed()));
+		onView(withId(R.id.main_menu_button_web))
+				.check(matches(isDisplayed()));
+		onView(withId(R.id.main_menu_button_upload))
+				.check(matches(isDisplayed()));
+
+		assertTrue(checkViewDirection(R.id.main_menu_button_continue));
+		assertTrue(checkViewDirection(R.id.main_menu_button_new));
+		assertTrue(checkViewDirection(R.id.main_menu_button_programs));
+		assertTrue(checkViewDirection(R.id.main_menu_button_help));
+		assertTrue(checkViewDirection(R.id.main_menu_button_web));
+		assertTrue(checkViewDirection(R.id.main_menu_button_upload));
+
+		assertTrue(checkTextDirection(getResourcesString(R.string.main_menu_continue)));
+		assertTrue(checkTextDirection(getResourcesString(R.string.main_menu_new)));
+		assertTrue(checkTextDirection(getResourcesString(R.string.main_menu_programs)));
+		assertTrue(checkTextDirection(getResourcesString(R.string.main_menu_help)));
+		assertTrue(checkTextDirection(getResourcesString(R.string.main_menu_web)));
+		assertTrue(checkTextDirection(getResourcesString(R.string.main_menu_upload)));
+
+		onView(withText(R.string.main_menu_programs))
```

Comment:
do not move through activites / perform actions you actually are not testing.
if you want to check the program menu activity, or other activites (like look list activity further down). write separate test classes for those and open them using the activity test rule.

Reply:
ok, I just wrote 2 test cases for MainMenuActivity and MyProjectsActivity. after the 5 Rtl PRs get merged, I think should I or someone else writes an RTL test cases for the whole App